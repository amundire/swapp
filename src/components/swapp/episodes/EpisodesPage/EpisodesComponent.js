import React from 'react';
import Episode from '../Episode/Episode';

const EpisodesComponent = ({data}) => {
  return (<section id="episodesList ">
    {data.allEpisodes.edges.map(({ node }) => <Episode id={node.id} image={node.image} title={node.title}/>)}
  </section>
  )
}

export default EpisodesComponent;