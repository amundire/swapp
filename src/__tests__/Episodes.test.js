import React from 'react';
import { shallow } from 'enzyme';
import EpisodesComponent from '../components/swapp/episodes/EpisodesPage/EpisodesComponent';

describe('<EpisodesPage />', () => {
    it('renders without crashing', () => {
        const data = {
            allEpisodes: {
                edges: [
                    {
                        node: {
                            id: 'films.1',
                            title: 'A New Hope',
                            episodeId: 4,
                            image: 'https://m.media-amazon.com/images/I/81r+LN-YReL._SS500_.jpg',
                            openingCrawl: 'It is a period of civil war.Rebel spaceships, strikingfrom a hidden base, have wontheir first victory againstthe evil Galactic Empire.During the battle, Rebelspies managed to steal secretplans to the Empires ultimate weapon, the DEATHSTAR, an armored spacestation with enough powerto destroy an entire planet.Pursued by the Empires sinister agents, PrincessLeia races home aboard herstarship, custodian of thestolen plans that can save herpeople and restorefreedom to the galaxy...',
                        },
                    },
                ],
            },
        },
        wrapper = shallow(<EpisodesComponent data={data} />);
        expect(wrapper).toMatchSnapshot();
    });
});
