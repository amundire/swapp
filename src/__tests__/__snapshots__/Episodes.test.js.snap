// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EpisodesPage /> renders without crashing 1`] = `
<section
  id="episodesList"
>
  <Episode
    id="films.1"
    image="https://m.media-amazon.com/images/I/81r+LN-YReL._SS500_.jpg"
    openingCrawl="It is a period of civil war.Rebel spaceships, strikingfrom a hidden base, have wontheir first victory againstthe evil Galactic Empire.During the battle, Rebelspies managed to steal secretplans to the Empires ultimate weapon, the DEATHSTAR, an armored spacestation with enough powerto destroy an entire planet.Pursued by the Empires sinister agents, PrincessLeia races home aboard herstarship, custodian of thestolen plans that can save herpeople and restorefreedom to the galaxy..."
    title="A New Hope"
  />
</section>
`;
